src/components/autocomplete/AutocompleteReducers.ts

```diff
@@ -0,0 +1,70 @@
+import * as _ from 'underscore';
+import {IReduxAction} from '../../utils/ReduxUtils';
+import {ListBoxActions} from '../listBox/ListBoxActions';
+import {AutocompleteActions, IAutocompletePayload} from './AutocompleteActions';
+
+export interface IAutocompleteState {
+    id: string;
+    open: boolean;
+    value: string;
+    active: number;
+}
+
+export const autocompleteInitialState: IAutocompleteState = {id: undefined, open: false, value: '', active: undefined};
+export const autocompleteCompositeInitialState: IAutocompleteState[] = [];
+
+export const autocompleteReducer = (state: IAutocompleteState = autocompleteInitialState, action: IReduxAction<IAutocompletePayload>): IAutocompleteState => {
+    if (state.id !== action.payload.id && action.type !== AutocompleteActions.add) {
+        return state;
+    }
+
+    switch (action.type) {
+        case AutocompleteActions.add:
+            return {
+                id: action.payload.id,
+                open: state.open,
+                value: state.value,
+                active: state.active,
+            };
+        case AutocompleteActions.toggle:
+            const open = !_.isUndefined(action.payload.open) ? action.payload.open : !state.open;
+            // if open was sent in the dispatch use the value. Otherwise toggle the property
+            return {...state, open, active: undefined};
+        case AutocompleteActions.setValue:
+            return {...state, open: true, value: action.payload.value};

```

Comment:
why does open become true on `setValue` ?

Reply:
I want to open the dropdown when the user type in the input

Reply:
I added a props an `open` in the payload to make it more obvious